protocols/gossipsub/src/protocol.rs

```diff
@@ -66,22 +66,22 @@ pub(crate) const FLOODSUB_PROTOCOL: ProtocolId = ProtocolId {
 pub struct ProtocolConfig {
     /// The Gossipsub protocol id to listen on.
     pub(crate) protocol_ids: Vec<ProtocolId>,
-    /// The maximum transmit size for a packet.
-    pub(crate) max_transmit_size: usize,
     /// Determines the level of validation to be done on incoming messages.
     pub(crate) validation_mode: ValidationMode,
+    /// Configurations for topics including max transmit size and mesh parameters.
+    pub(crate) config: TopicConfigs,
```

Comment:
Don't think we need the whole config here right? Just like a HashMap<TopicHash, usize> ? i.e the `max_transmit_size_for_topic` ?